src/ucp/dt/datatype_iter.c

```diff
@@ -115,7 +119,7 @@ void ucp_datatype_iter_iov_mem_dereg(ucp_datatype_iter_t *dt_iter)
     ucp_mem_h *memh = dt_iter->type.iov.memh;
     size_t iov_index, length;
 
-    if ((memh == NULL) || ucp_memh_is_user_memh(*memh)) {
+    if ((memh == NULL) || (*memh == NULL) || ucp_memh_is_user_memh(*memh)) {
```

Comment:
pls check when we can have `memh != NULL` but `*memh ==NULL`

Reply:
With the new `iov_count` checks it seems like there is no such case. Thanks!

```suggestion
    if ((memh == NULL) || ucp_memh_is_user_memh(*memh)) {
```